easybuild/framework/easyconfig/default.py

```diff
@@ -92,6 +92,7 @@
     'checksums': [[], "Checksums for sources and patches", BUILD],
     'configopts': ['', 'Extra options passed to configure (default already has --prefix)', BUILD],
     'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with (if supported)", BUILD],
+    'data_sources': [[], "List of source files for data", BUILD],
```

Comment:
Is there a need to separate `data_sources` from `sources`?

Reply:
data sources can be very big so i think it's good to at least have an option to separate them.

we can set `data_sources` equal to `sources` by default, would you prefer that?

Reply:
done in https://github.com/easybuilders/easybuild-framework/pull/4474/commits/2e57ad18d356ab0c3d23d5bd8c9027f62699c54b https://github.com/easybuilders/easybuild-framework/pull/4474/commits/8dbf7d3c61b491dd474ea55d69ad33ce40c6d9ef

Reply:
hm, i see now that i misunderstood you.
the separate parameter `data_sources` was suggested by @boegel to make it clear they are different from software sources.
there is no real need for it, it's just cosmetic. i'm not sure if it's a good idea, happy to revert if you prefer.

Reply:
as discussed during the EUM, we should keep `data_sources` to allow installing software and datasets in a single easyconfig (e.g. using components) if we want to add support for this later on. this PR does support this, as it requires substantive changes and i'm unsure how useful it is. i've modified this PR to make it easier to implement support for it if/when desired.